### DISCUSSION TITLE
`custom_section` highlights

### DISCUSSION THREAD

**Discussion by olimorris** [utilyre/barbecue.nvim#67] [Ideas]
Unless I've missed it, would be awesome to have a section in the theme for the custom_section.

**Comment by utilyre** [comment]
`custom_section` is supposed to have it's highlights inlined. (See also #49)

From `:h 'statusline'`

> 	\# -   Set highlight group.  The name must follow and then a # again.
>	      Thus use %#HLname# for highlight group HLname.  The same
>	      highlighting is used, also for the statusline of non-current
>	      windows.

**NOTE**: I'll probably have the `custom_section` return a list like table of `[text, highlight]` elements in the next major/minor version.

**Comment by utilyre** [comment]
I don't know why github doesn't show mentioned discussions, anyways see this 83d2591.

**Reply by olimorris** [reply]
Just to check I'm understanding the API correctly, this should work (where VimLogo is a highlight group I've defined):

```lua
custom_section = function() return "  ", "VimLogo" end
```

Unfortunately the highlight doesn't seem to be applied unlike when I do `custom_section = function() return "%#VimLogo#  " end`

**Reply by utilyre** [reply]
You've got to return a list of `[text, highlight]` tuples. E.g.

```lua
return {
  { " " },
  { "", "VimLogo" },
  { " " },
}
```

BTW, are you defining `VimLogo` highlight elsewhere? Because it's not defined by neither Neovim or my colorscheme.

**Reply by olimorris** [reply]
Amazing thank you. And yep it's defined in my colorscheme.
